### PR TITLE
Add competitor comparison table

### PR DESCRIPTION
**Pull request by guimachiavelli**
Due to issues with how github handles PRs based on branches deleted by the original author, it is impossible to merge #1336 now.

This PR simply adds the previously accepted changed on #1336.

Credits go to @ferdi05 for the table and @maryamsulemani97 for proofreading and review.

### CODE REVIEW

File: learn/what_is_meilisearch/comparison_to_alternatives.md [hunk 1]
```diff
@@ -6,15 +6,148 @@
...
+|---|:----:|:----:|:-----:|:----:|
+| Source code licensing | [MIT](https://choosealicense.com/licenses/mit/) <br> (Fully open-source) |  Closed-source  | [GPL-3](https://choosealicense.com/licenses/gpl-3.0/) <br> (Fully open-source) | SSPL <br>([Not open-source](https://opensource.org/node/1099))  |
+| Built with | Rust <br> [Check out why we believe in Rust](https://www.abetterinternet.org/docs/memory-safety/). | C++ | C++ | Java |
+| Data storage | Disk with Memory Mapping -- Not limited by RAM | Limited by RAM | Limited by RAM | Disk with RAM cache |
+
+### Features
+
+#### Integrations and SDKs
+
+We list only officially supported libraries by the internal teams of the different projects.
```
Comment (guimachiavelli) [inline learn/what_is_meilisearch/comparison_to_alternatives.md:35]:
```suggestion
Note: we are only listing libraries officially supported by the internal teams of each different search engine.
```

File: learn/what_is_meilisearch/comparison_to_alternatives.md [hunk 2]
```diff
@@ -6,15 +6,148 @@
...
+|   | MeiliSearch | Algolia | Typesense | Elasticsearch |
+|---|:---:|:----:|:---:|:---:|
+| Placeholder search | ✅ | ✅ | ✅ | ✅ |
+| Multi-index search | Early 2022 | ✅ | ✅ | ✅ |
+| Exact phrase search | ✅ | ✅ | ❌ | ✅ |
+| Geo search |  ✅  | ✅ | ✅ | ✅ |
+| Sort by  |  ✅  | 🔶 <br> Limited to one `sort_by` rule per index. Index have to be duplicated for each sort field and sort order | 🔶 <br> Does not support sort on string field | ✅ |
+| Filtering |  ✅ <br> Support complex filter queries with an SQL-like syntax.  | 🔶 <br> Does not support `OR` operation across multiple fields | 🔶 <br> Does not support `OR` operation across multiple fields | ✅ |
+| Faceting |  ✅ | ✅ | ✅ | ✅ |
+| Facet value searches | ❌ | ✅ | ✅ | ❌ |
```
Comment (guimachiavelli) [inline learn/what_is_meilisearch/comparison_to_alternatives.md:103]:
```suggestion
```

File: learn/what_is_meilisearch/comparison_to_alternatives.md [hunk 3]
```diff
@@ -6,15 +6,148 @@
...
+
+##### Search
+
+|   | MeiliSearch | Algolia | Typesense | Elasticsearch |
+|---|:---:|:----:|:---:|:---:|
+| Placeholder search | ✅ | ✅ | ✅ | ✅ |
+| Multi-index search | Early 2022 | ✅ | ✅ | ✅ |
+| Exact phrase search | ✅ | ✅ | ❌ | ✅ |
+| Geo search |  ✅  | ✅ | ✅ | ✅ |
+| Sort by  |  ✅  | 🔶 <br> Limited to one `sort_by` rule per index. Index have to be duplicated for each sort field and sort order | 🔶 <br> Does not support sort on string field | ✅ |
```
Comment (guimachiavelli) [inline learn/what_is_meilisearch/comparison_to_alternatives.md:100]:
```suggestion
| Sort by  |  ✅  | 🔶 <br> Limited to one `sort_by` rule per index. Indexes have to be duplicated for each sort field and sort order | 🔶 <br> Does not support sort on string field | ✅ |
```

**Comment by ferdi05** [comment]
To respect consistency with the documentation, I would go with `Faceted search`.

File: learn/what_is_meilisearch/comparison_to_alternatives.md [hunk 4]
```diff
@@ -6,15 +6,147 @@
...
+
+|   | MeiliSearch | Algolia | Typesense | Elasticsearch |
+|---|:---:|:----:|:---:|:---:|
+| Placeholder search | ✅ | ✅ | ✅ | ✅ |
+| Multi-index search | Early 2022 | ✅ | ✅ | ✅ |
+| Exact phrase search | ✅ | ✅ | ❌ | ✅ |
+| Geo search |  ✅  | ✅ | ✅ | ✅ |
+| Sort by  |  ✅  | 🔶 <br> Limited to one `sort_by` rule per index. Indexes have to be duplicated for each sort field and sort order | 🔶 <br> Does not support sort on string field | ✅ |
+| Filtering |  ✅ <br> Support complex filter queries with an SQL-like syntax.  | 🔶 <br> Does not support `OR` operation across multiple fields | 🔶 <br> Does not support `OR` operation across multiple fields | ✅ |
+| Faceting |  ✅ | ✅ | ✅ | ✅ |
```
Comment (ferdi05) [inline learn/what_is_meilisearch/comparison_to_alternatives.md]:
```suggestion
| Faceted search |  ✅ | ✅ | ✅ | ✅ |
```

Comment (guimachiavelli) [inline learn/what_is_meilisearch/comparison_to_alternatives.md:102]:
```suggestion
| Faceted search |  ✅ | ✅ | ✅ | ✅ |
```

**Comment by guimachiavelli** [comment]
Good catch, thanks!

**Review by ferdi05 — APPROVED** [review]

**Review by ferdi05 — APPROVED** [review]
LGTM

**Comment by guimachiavelli** [comment]
bors merge

**Comment by curquiza** [comment]
Waaaah huge work here!!!

**Comment by guimachiavelli** [comment]
Indeed! Huge thanks to @ferdi05 for all the work he put in creating this!

**Pull request merged**